storage/src/tests/persistence/filestorage/service_layer_host_info_reporter_test.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include <vespa/storage/persistence/filestorage/service_layer_host_info_reporter.h>
+#include <tests/common/hostreporter/util.h>
+#include <tests/common/testnodestateupdater.h>
+#include <vespa/vespalib/data/slime/slime.h>
+#include <vespa/vespalib/io/fileutil.h>
+#include <vespa/vespalib/stllike/asciistream.h>
+#include <vespa/vespalib/gtest/gtest.h>
+#include <iostream>
+
+namespace storage {
+
+namespace {
+
+double
+get_usage_element(const vespalib::Slime& root, const vespalib::string& label)
+{
+    return root.get()["content-node"]["resource-usage"][label]["usage"].asDouble();
+}
+
+}
+
+struct ServiceLayerHostInfoReporterTest : ::testing::Test {
+
+    TestNodeStateUpdater         _state_manager;
+    ServiceLayerHostInfoReporter _reporter;
+
+    ServiceLayerHostInfoReporterTest();
+    ~ServiceLayerHostInfoReporterTest();
+
+    void notify(double disk_usage, double memory_usage)
+    {
+        auto& listener = static_cast<spi::IResourceUsageListener&>(_reporter);
+        listener.update_resource_usage(spi::ResourceUsage(disk_usage, memory_usage));
+    }
+
+    size_t requested_almost_immediate_replies() { return _state_manager.requested_almost_immediate_node_state_replies(); }
+    std::vector<double> get_old_usage() {
+        auto &old_resource_usage = _reporter.get_old_resource_usage();
+        return std::vector<double>{ old_resource_usage.get_disk_usage(), old_resource_usage.get_memory_usage() };
```

Comment:
Wouldn't it be better to just return the storage::spi::ResourceUsage class? And add an _operator==_ to that class and an external _operator<<_ function. This applies to the two functions below as well.